### ISSUE TITLE
Empty accessory context

### ISSUE THREAD

**Issue by jamesrodda** [homebridge/homebridge#2566]
I'm the author of [homebridge-mihomegateway](https://github.com/jamesrodda/homebridge-mihomegateway) and I've got an error that suddenly cropped up on my local instance of homebridge.

When an accessory is added, the my raw platform device (i.e. the device that is returned from the call to the actual gateway API) is stored under `accessory.context.device`. This means when `configureAccessory` is called on subsequent loads, the full device info is available and the device can be handled.

However, at some point, the context has been lost and is just an empty object - this means I'm unable to recreate the accessory from the cache. I can see this in the `cachedAccessories` file.

Firstly, what might have caused this to have happened? Are there any events which clear the context of an accessory?

Secondly, is it possible to handle such a case? Homebridge is trying to re-add an accessory from cache - do I just ignore it and allow `discoverDevices` to re-add it fully? Would there be UUID conflicts in this case?

**Comment by Supereg** [comment]
We're you running any beta versions of homebridge? This is the first time hearing of a data loss like this 🤔

> Secondly, is it possible to handle such a case?

Best case would be to just clear the entires from the `cachedAccessories` file, or you programmatically call `unregisterPlatformAccessories` for those which can't be recreated.

**Comment by oznu** [comment]
@jamesrodda are you testing while using `npm run watch`? 

If so nodemon is probably sending SIGKILL which does not give Homebridge a chance to save the cache on shutdown. If you edit the `nodemon.json` file to look like this the cache should be saved when Homebridge restarts:

```diff
  {
    "watch": [
      "src"
    ],
    "ext": "ts",
    "ignore": [],
    "exec": "tsc && homebridge -I -D",
+   "signal": "SIGTERM"
  }
```

**Comment by oznu** [comment]
I've updated the file in the template:

https://github.com/homebridge/homebridge-plugin-template/blob/homebridge-beta-changes/nodemon.json

**Comment by jamesrodda** [comment]
@Supereg I haven't run any beta versions at all, either during dev or "live".

I tried calling `unregisterPlatformAccessories` but gt an error saying it didn't exist - presumably because, at this point when I'm recreating from cache, they don't really exist yet?

I've batted my `cachedAccessories` file to resolve this as I needed t get it up and running but it's easy to reproduce so will try a few other things to try an recover in this case. Would be keen to figure out what's actually happened here though.

@oznu thanks, I'll update `nodemon.json` but this happened on my "live" instance (Raspberry Pi running your docker image) so nodemon doesn't even feature here...
